JUnit output format fails: invalid memory address

Hey, I'm trying to use the JUnit output, but for some reason, it fails with a memory error.
Tried other formats and those are working fine.


```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf2245c]

goroutine 1 [running]:
github.com/styrainc/regal/pkg/reporter.JUnitReporter.Publish({{0x1943a20?, 0xc0000a6028?}}, {0xc000b08ef8?, 0xc000a2e5c0?}, {0x0, 0x0, 0x0, 0x0, {0xc000245008, 0x28, ...}, ...})
        /home/runner/work/regal/regal/pkg/reporter/reporter.go:525 +0x85c
github.com/styrainc/regal/cmd.lint({0xc000896a80, 0x1, 0x3}, 0xc000c7d680)
        /home/runner/work/regal/regal/cmd/lint.go:341 +0x17f4
github.com/styrainc/regal/cmd.init.4.func2({0xc000896a80, 0x1, 0x3})
        /home/runner/work/regal/regal/cmd/lint.go:109 +0xb4
github.com/styrainc/regal/cmd.init.4.wrapProfiling.func3(0xc0002bf208, {0xc000896a80, 0x1, 0x3})
        /home/runner/work/regal/regal/cmd/profiling.go:16 +0x733
github.com/spf13/cobra.(*Command).execute(0xc0002bf208, {0xc000896a20, 0x3, 0x3})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0x222b560)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(0x0?)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041 +0x13
main.main()
        /home/runner/work/regal/regal/main.go:16 +0x2b
```

command used:

`` 
regal lint --format junit ./policies/claims
``

--- a/internal/capabilities/capabilities_integration_test.go
+++ b/internal/capabilities/capabilities_integration_test.go
@@ -18,7 +18,7 @@ func TestLookupFromURL(t *testing.T) {
 
 	caps, err := Lookup(
 		context.Background(),
-		"https://raw.githubusercontent.com/open-policy-agent/opa/refs/heads/main/v1/capabilities/v0.55.0.json",
+		"https://raw.githubusercontent.com/open-policy-agent/opa/main/capabilities/v0.55.0.json",
 	)
 	if err != nil {
 		t.Errorf("unexpected error from Lookup: %v", err)

--- a/pkg/reporter/reporter.go
+++ b/pkg/reporter/reporter.go
@@ -511,6 +511,11 @@ func (tr JUnitReporter) Publish(_ context.Context, r report.Report) error {
 		}
 
 		for _, violation := range violationsPerFile[file] {
+			text := ""
+			if violation.Location.Text != nil {
+				text = strings.TrimSpace(*violation.Location.Text)
+			}
+
 			testsuite.AddTestcase(junit.Testcase{
 				Name:      fmt.Sprintf("%s/%s: %s", violation.Category, violation.Title, violation.Description),
 				Classname: violation.Location.String(),
@@ -522,7 +527,7 @@ func (tr JUnitReporter) Publish(_ context.Context, r report.Report) error {
 						violation.Description,
 						violation.Category,
 						violation.Location.String(),
-						strings.TrimSpace(*violation.Location.Text),
+						text,
 						getDocumentationURL(violation)),
 				},
 			})

--- a/pkg/reporter/reporter_test.go
+++ b/pkg/reporter/reporter_test.go
@@ -331,6 +331,17 @@ func TestJUnitReporterPublishNoViolations(t *testing.T) {
 	}
 }
 
+func TestJUnitReporterPublishViolationWithoutText(t *testing.T) {
+	t.Parallel()
+
+	var buf bytes.Buffer
+	if err := NewJUnitReporter(&buf).Publish(context.Background(), report.Report{
+		Violations: []report.Violation{{Title: "no-text"}},
+	}); err != nil {
+		t.Fatal(err)
+	}
+}
+
 func MustReadFile(t *testing.T, path string) string {
 	t.Helper()
 

